### PR TITLE
Feature/fix a bunch of warnings

### PR DESCRIPTION
**Pull request by johandahlberg**
## Description

Fixes a bunch of warnings.

Fixes: EXE-1243, EXE-1203, EXE-1224

## Type of change

Cleaning up warnings.

## How Has This Been Tested?

Tested with unit test suite.

## PR checklist:

- [ ] This comment contains a description of changes (with reason).
- [ ] My code follows the style guidelines of this project
- [ ] Make sure your code lints, is well-formatted and type checks
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If you've added a new stage - have you followed the pipeline CLI conventions in the [USAGE guide](../USAGE.md)
- [ ] [README.md](./README.md) is updated
- [ ] If a new tool or package is included, update poetry.lock, [the conda recipe](../conda-recipe/pixelator/meta.yaml) and [cited it properly](../CITATIONS.md)
- [ ] Include any new [authors/contributors](../AUTHORS.md)
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and documentation and corrected any misspellings
- [ ] Usage Documentation is updated
- [ ] If you are doing a [release](../RELEASING.md#Releasing), or a significant change to the code, update [CHANGELOG.md](../CHANGELOG.md)

### CODE REVIEW

**Comment by johandahlberg** [comment]
@ambarrio You can ignore the failing linting here. It's probably some inconsistency between the version of black used in CI and the one I have locally. I will look at resolving that, and submit it as a separate formatting-only PR so we don't have to deal with that in this review. Mixing lots of formatting changes and actual code changes makes for confusing reviews in my opinion.

File: src/pixelator/pixeldataset/utils.py [hunk 1]
```diff
@@ -55,7 +56,11 @@ def update_metrics_anndata(adata: AnnData, inplace: bool = True) -> Optional[Ann
     df = adata.to_df()
 
     # update the var layer (antibody metrics)
-    adata.var["antibody_count"] = df.sum().astype(int)
+    # we ignore the warning here, since we actually want to force and update of the
+    # `adata.var` data frame.
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", category=ImplicitModificationWarning)
+        adata.var["antibody_count"] = df.sum().astype(int)
```
Comment (ambarrio) [inline src/pixelator/pixeldataset/utils.py:63]: Does it only applies to L63 or also to the next two lines L64-65?

**Review by ambarrio — COMMENTED** [review]
A small comment only

File: src/pixelator/pixeldataset/utils.py (hunk 1 above)
Comment (johandahlberg) [reply src/pixelator/pixeldataset/utils.py:63]: Only the lines which are encapsulated by the context manager, i.e. only L63.

**Review by ambarrio — APPROVED** [review]

**Pull request merged**